### PR TITLE
Web resolution

### PR DESCRIPTION
**Pull request by oa495**
- allow variants to have distribute prop (can choose which variant goes to sprite)
- remove resolution from name

### CODE REVIEW

File: packages/@icon-magic/distribute/src/utils.ts [hunk 1]
```diff
@@ -61,4 +57,13 @@ export function getIconFlavorsByType(icon: Icon, type: FlavorType): Asset[] {
...
-}
\ No newline at end of file
+}
+
+/**
+ * Strips resolution from icon name
+ * @param iconName name to strip resolution from
+ * @returns a string with the resolution stripped
+ */
+export function removeResolutionFromName(iconName: string): string {
```
Comment (oa495) [inline packages/@icon-magic/distribute/src/utils.ts]: add to svg-generate? Web only needs one resolution as does the sprite

File: packages/@icon-magic/config-reader/src/schemas/config-schema.ts [hunk 2]
```diff
@@ -62,6 +57,16 @@ const distributeConfigProperties = {
   }
 };
 
+const variantProperties = {
+  name: { type: 'string' },
+  path: { type: 'string', minLength: 1 },
+  distribute: {
+    type: 'object',
```
Comment (ramitha) [inline packages/@icon-magic/config-reader/src/schemas/config-schema.ts]: Can be null too

File: packages/@icon-magic/distribute/src/distribute-svg.ts [hunk 3]
```diff
@@ -27,30 +31,45 @@ export async function distributeSvg(
...
-      // Just copy the files to the output
-      await copyIconAssetSvgs(icon.iconName, assets, outputPath);
-    } else {
+    );
+
+    if (iconHasSpriteConfig || assetsToAddToSprite.length) {
+      if (assetsToAddToSprite.length && !iconHasSpriteConfig) {
+        assets = assetsToAddToSprite;
+      }
+      // TO-DO: Handle case where icon config enables sprite but one asset disables
```
Comment (ramitha) [inline packages/@icon-magic/distribute/src/distribute-svg.ts]: Not sure we need this as it's just another way to add a fixed set of assets to the sprite and not all. We can get rid of the TODO

File: packages/@icon-magic/distribute/test/index-test.ts [hunk 4]
```diff
@@ -180,4 +193,23 @@ describe('distribute works as expected', function() {
...
+      svgIDs.forEach(id => {
+        assert.ok(
+          doc.getElementById(id),
+          'puts the right variants into sprite'
+        );
+      });
+    } catch (err) {
+      assert.ok(false, err);
+    }
+  });
```
Comment (ramitha) [inline packages/@icon-magic/distribute/test/index-test.ts]: Add a test for stripping the resolution from the name too?

Comment (oa495) [reply packages/@icon-magic/distribute/test/index-test.ts]: it's tested when the IDs are checked in line 206.

File: packages/@icon-magic/config-reader/src/schemas/config-schema.ts [hunk 5]
```diff
@@ -41,6 +36,9 @@ const buildConfigProperties = {
 };
 
 const distributeConfigProperties = {
+  filterByVariants: {
```
Comment (ramitha) [inline packages/@icon-magic/config-reader/src/schemas/config-schema.ts]: Can we have it both here and under svg if we're now going to contain it only for svg distribute? We can assume that if it's here, it applies to all types of distribution

Comment (oa495) [reply packages/@icon-magic/config-reader/src/schemas/config-schema.ts]: updated.

**Pull request merged**